website/docs/tutorials/guides/testing.md

```diff
@@ -2,10 +2,185 @@
+
+```cairo title="Fixtures and type definitions"
+const EXTERNAL_CONTRACT_ADDRESS = 0x3fe90a1958bb8468fb1b62970747d8a00c435ef96cda708ae8de3d07f1bb56b
+
+struct Point:
+    member x : felt
+    member y : felt
+end
+
+@contract_interface
+namespace ITestContract:
+    func get_felt() -> (res : felt):
+    end
+
+    func get_array() -> (res_len : felt, res : felt*):
+    end
+
+    func get_struct() -> (res : Point):
+    end
+end
+```
+
+##### Felt
+
+```cairo title="mocked_call returns a felt"
+@view
+func test_mock_call_returning_felt{syscall_ptr : felt*, range_check_ptr}():
+  tempvar external_contract_address = EXTERNAL_CONTRACT_ADDRESS
+
+  %{ mock_call(ids.external_contract_address, "get_felt", [42]) %}
+  let (res) = ITestContract.get_felt(EXTERNAL_CONTRACT_ADDRESS)
+
+  assert res = 42
+  return ()
+end
+```
+
+##### Array
+
+To mock a function returning an array, provide data in the following format to `ret_data`:
+
+```python title="Python representation of a Cairo array"
+[n, value_1, value_2, ..., value_n]
+```
+
+```cairo title="mocked_call returns an array"
+@view
+func test_mock_call_returning_array{syscall_ptr : felt*, range_check_ptr}():
+  tempvar external_contract_address = EXTERNAL_CONTRACT_ADDRESS
+
+  %{ mock_call(ids.external_contract_address, "get_array", [1, 42]) %}
+  let (res_len, res_arr) = ITestContract.get_array(EXTERNAL_CONTRACT_ADDRESS)
+
+  assert res_arr[0] = 42
+  return ()
+end
+```
+
+##### Struct
+
+```cairo title="mocked_call returns a struct"
+@view
+func test_mock_call_returning_struct{syscall_ptr : felt*, range_check_ptr}():
+  tempvar external_contract_address = EXTERNAL_CONTRACT_ADDRESS
+
+  %{ mock_call(ids.external_contract_address, "get_struct", [21,37]) %}
+  let (res_struct) = ITestContract.get_struct(EXTERNAL_CONTRACT_ADDRESS)
+
+  assert res_struct.x = 21
+  assert res_struct.y = 37
+  return ()
+end
+```
+
+### `clear_mock_call`
+
+```python
+def clear_mock_call(contract_address: int, fn_name: str) -> None: ...
+```
+
+Removes a mocked call specified by a function name (`fn_name`) of a contract with an address (`contract_address`).
+
+### `expect_revert`
+
+```python
+def expect_revert() -> None: ...
+```
+
+If a code beneath `expect_revert` triggers revert, a test will pass.
+
+```cairo title="This test passes despite using a random contract address."
+@external
+func test_call_to_non_existing_contract{syscall_ptr : felt*, range_check_ptr}():
+    alloc_locals
+
+    local contract_a_address : felt
+    %{ ids.contract_a_address = 0101010101010101010101010 %}
+
+    %{ expect_revert() %}
+    BasicContract.increase_balance(contract_address=contract_a_address, amount=3)
+    return ()
+end
+```
+
+### `deploy_contract`
+
+```python
+def deploy_contract(contract_path: str) -> Contract:
+```
```

Comment:
Can we link the related section here?

Reply:
Did you merge your testing docs to master?

Reply:
I'll just copy what you did in that PR. I tried to pull the branch, but there are too many changes.

Reply:
If you copy integration testing here, could you please move it on the top of the list? I think this cheat code is the most important one